File: volunteers/src/renderer/functions/worker.js
import baseURL from "./baseURL";
import axios from "axios";

export default function (phase, step_id, hash, context) {
    const response = await axios.post(
        `${baseURL}/worker/submit-result`,
        {
            step_id: step_id,
            phase: phase,
            result_file_id: hash
        },
        {
            headers: { 'Content-Type': 'application/json' }
        }
    );
    if (response.STATUS == "OK") {
        return [true, undefined]
    }
    else if (response.STATUS == "FAIL") {
        return [false, response.MSG]
    }
    return [false, "Problem with API call. Contact Developers."]
}

File: runtimes/python/worker.py
import imp
import os
import requests
import os
from python.main import sendHash

phase = os.getenv("PHASE")
step_id = ""
hash = sendHash()
url = 'http://127.0.0.1:5000/worker/submit-result'
params = {"phase" : phase, "step_id": step_id, "result_file_id": hash}
r = requests.post(url, params=params)
